publication-event-dtopublicationdoi-producer/src/test/java/no/unit/nva/doi/event/producer/DynamoDbEventConducerTest.java

```diff
@@ -0,0 +1,92 @@
+package no.unit.nva.doi.event.producer;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.file.Path;
+import no.unit.nva.events.handlers.EventParser;
+import nva.commons.utils.IoUtils;
+import nva.commons.utils.JsonUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class DynamoDbEventConducerTest {
+
+    private static final String WILDCARD = "*";
+    public static final String EXAMPLE_NAMESPACE = "https://example.net/unittest/namespace/";
+    public static final String DOI_PUBLICATION_TYPE = "doi.publication";
+    private static final String DYNAMODB_STREAM_EVENT_OLD_AND_NEW_PRESENT_DIFFRENT =
+        "dynamodbevent_old_and_new_present_different.json";
+    private static final String DYNAMODB_STREAM_EVENT_OLD_AND_NEW_PRESENT_EQUAL =
+        "dynamodbevent_old_and_new_present_equal.json";
+    private static final String DYNAMODB_STREAM_EVENT_OLD_ONLY = "dynamodbevent_old_only.json";
+    private AppConfig environmentMock;
+    private static final String DYNAMODB_STREAM_EVENT = "dynamodbevent.json";
+    private static ObjectMapper objectMapper = JsonUtils.objectMapper;
+    private DynamoDbEventConducer handler;
+    private Context context;
+
+    /**
+     * Setting up test environment.
+     */
+    @BeforeEach
+    public void setUp() {
+        handler = new DynamoDbEventConducer(EXAMPLE_NAMESPACE);
+        context = mock(Context.class);
+    }
+
+    @Test
+    void processInput() throws IOException {
+        var eventFile = IoUtils.stringFromResources(Path.of(DYNAMODB_STREAM_EVENT));
+        DynamodbEvent event = objectMapper.readValue(eventFile, DynamodbEvent.class);
+        var eventBridgeEvent = new EventParser<DynamodbEvent>(
+            eventFile).parse(DynamodbEvent.class);
+        var actual = handler.processInput(event, eventBridgeEvent, context);
+
+        assertThat(actual.getType(), is(equalTo(DOI_PUBLICATION_TYPE)));
+        assertThat(actual.getItems(), hasSize(1));
+    }
+
+    @Test
+    void noNewImageWhileProcessingInputThenInEffectiveChangeIsNotIncluded() throws JsonProcessingException {
```

Comment:
does this mean processInputFiltersOutDynamoStreamEventsWithNoNewImage?

Reply:
Given `noNewImage` While `processingInput` Then `InEffectiveChangeIsNotIncluded` . 

Or do you want:

While `processingInput` Then `InEffectiveChangeIsNotIncluded` given `NoNewImage` ?

Reply:
For readability and for better understanding of the intention of the test we are trying to follow the following convetion:
 methodName(Returns/Throws/Does)SomethingWhen(Input/Condition)IsSomething. 

"While" here is confusing because it has the meaning of "during". We do not care what the function does while it is running. We care about the output. Also, if we do not follow the same convention, then the test becomes harder to understand. 

The target is that the method name should as close to a real sentence as possible. For example, 
"processInput filters out dynamo-stream-events with no new image" is clear about the function that is testing (processInput), the input (dynamo-stream-events) and what it does with them (filters out)


"no new Image while processing input then in effective change is not included"  is not clear about the input (the input is not a new image) the name is implied, and the notion of an "ineffective change" is defined in your code. Basically, It not clear that we are talking about a dynamo stream event and that the ineffective change is the fact that there is no new image. 
In other words one has to read the code, spot the method ineffective change, understand what is happening there and come back here to read the test.  The test however tests something very simple: Are dynamo-stream-events with no new image excluded or not?

Reply:
Thanks for sharing your thoughts, will update. Makes it easier to understand how you want em written. :)